src/test/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardErrorEmailServiceImplTest.java

```diff
@@ -1,40 +1,132 @@
 package edu.cornell.kfs.fp.batch.service.impl;
 
-import java.util.ArrayList;
-
-import org.kuali.kfs.fp.document.ProcurementCardDocument;
+import junit.framework.TestCase;
 import org.kuali.kfs.sys.ConfigureContext;
-import org.kuali.kfs.sys.context.KualiTestBase;
-import org.kuali.kfs.sys.context.SpringContext;
-import org.kuali.kfs.sys.context.TestUtils;
-import org.kuali.rice.krad.service.MailService;
-
-import edu.cornell.kfs.fp.batch.service.ProcurementCardErrorEmailService;
+import org.kuali.kfs.sys.service.impl.DevelopmentMailServiceImpl;
+import org.kuali.rice.coreservice.api.parameter.Parameter;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
 
+import java.util.ArrayList;
+import java.util.Collection;
 
 @ConfigureContext
-public class ProcurementCardErrorEmailServiceImplTest extends KualiTestBase {
-	private ProcurementCardErrorEmailService procurementCardErrorEmailService;	
-	private MailService mailService;
-	
-	private static String updateParm = "update KRCR_PARM_T SET  from GL_RVRSN_CTGRY_AMT_T";
-	
+public class ProcurementCardErrorEmailServiceImplTest extends TestCase {
+	private ProcurementCardErrorEmailServiceImpl procurementCardErrorEmailService;
+
 	@Override
     protected void setUp() throws Exception {
         super.setUp();
-        procurementCardErrorEmailService = SpringContext.getBean(ProcurementCardErrorEmailService.class);       
-        mailService = SpringContext.getBean(MailService.class);
+        procurementCardErrorEmailService = new ProcurementCardErrorEmailServiceImpl();
+		procurementCardErrorEmailService.setMailService(new DevelopmentMailServiceImpl());
+		procurementCardErrorEmailService.setParameterService(new MockParameterService());
     }
 	
 	public void testProcurementCardErrorEmailService(){
-		TestUtils.setSystemParameter(ProcurementCardDocument.class, "PCARD_UPLOAD_ERROR_EMAIL_ADDR", "fake@cornell.edu");
-
 		ArrayList<String> errorMessages = new ArrayList<String>();
 		errorMessages.add("TEST ERROR FOR PROCUREMENT CARD");
 		procurementCardErrorEmailService.sendErrorEmail(errorMessages);
-
-		
 	}
     
+	private class MockParameterService implements ParameterService {
```

Comment:
Would it make more sense to define some of these "mock" services in their own separate files instead, to make them reusable across multiple tests?